srcjs/init_shiny.js

```diff
@@ -414,7 +414,15 @@ function initShiny() {
   initialValues['.clientdata_url_hostname'] = window.location.hostname;
   initialValues['.clientdata_url_port']     = window.location.port;
   initialValues['.clientdata_url_pathname'] = window.location.pathname;
+
+  // Send initial URL search (query string) and update it if it changes
   initialValues['.clientdata_url_search']   = window.location.search;
+
+  // on popstate
+  window.onpopstate = history.onpushstate = function(e) {
```

Comment:
If possible I'd prefer not `onpopstate`/`onpushstate`, but `addEventListener("popstate", ...)` (or preferably, the jQuery equivalent). I don't know if there's something specific about pop/pushstate but in general you want to avoid the attribute-assignment style of event subscribing, since you could be overwriting a previous handler and are also susceptible to someone later overwriting yours.